common/lockmanager_test.go

```diff
@@ -0,0 +1,318 @@
+package common
+
+import (
+    "sync"
+    "testing"
+    "time"
+
+    "github.com/edulinq/autograder/config"
+)
+
+func TestLockBase(test *testing.T) {
+    key1 := "testkey1";
+
+    Lock(key1);
+
+    err := Unlock(key1);
+    if (err != nil) {
+        test.Fatalf("Failed to unlock.");
+    }
+}
+
+func TestUnlockingAnUnlockedLock(test *testing.T) {
+    key1 := "testkey1";
+
+    Lock(key1);
+
+    err := Unlock(key1);
+    if (err != nil) {
+        test.Fatalf("Failed to unlock.");
+    }
+
+    err = Unlock(key1);
+    if (err == nil) {
+        test.Fatalf("Failed to return an error when unlocking a lock that is already unlocked.");
+    }
+}
+
+func TestUnlockingKeyThatDoesntExist(test *testing.T) {
+    doesNotExistKey := "dne";
+
+    err := Unlock(doesNotExistKey);
+    if (err == nil) {
+        test.Fatalf("Failed to return an error when unlocking a key that does not exist.");
+    }
+}
+
+func TestLockUnlockDifferentKeysOneThread(test *testing.T) {
+    key1 := "testkey1";
+    key2 := "testkey2";
+    var verifyWaitGroup sync.WaitGroup;
+    var threadWaitGroup sync.WaitGroup;
+
+    // Lock key1 for the first time.
+    Lock(key1);
+
+    threadWaitGroup.Add(1);
+    verifyWaitGroup.Add(1);
```

Comment:
Consider more expressive names for your wait groups that are special for each test.
Like: threadCompletionBlock and key2LockedBlock.

This is not a required comment, but something you can decide.